### DISCUSSION TITLE
double click on inline buttons

### DISCUSSION THREAD

**Discussion by Javad-Smn** [telegraf/telegraf#1394] [Q&A]
Hello,
I made inline menu with telegram-keyboard library.
I have a button in every step that will back to pervious step.
Its action is when user clicked on it will remove current menu( deleteMessage() method ) and will print pervious menu.
But if user double click on button i will receive this error: **'Bad Request: message to delete not found'**.
I also have this problem with the other buttons.
Is there any solution to fix this problem?

**Comment by piyushburate** [comment]
**Explanation:**
This is not a problem, its just the behaviour of the Inline Buttons. Every time user click the button it send the callback query to handle. So if 1st clicked callback query completes its task like deleting message then 2nd clicked callback dont have that message to delete so it gives an error, i.e. Message to delete not found.

**Solution:**
Just wrap your whole code in callback query handler in try, catch block. so even if 2nd click gives error your bot working wont stop.

**Comment by hypernova7** [comment]
It's a normal behavior of deteleteMessage. Use try catch

```js
try {
  await ctx.deleteMessage()
} catch {}
```

That will prevent the error from appearing when trying to delete the message twice.
